src/workerd/io/actor-sqlite.c++

```diff
@@ -18,8 +36,11 @@ ActorSqlite::ActorSqlite(kj::Own<SqliteDatabase> dbParam,
       commitCallback(kj::mv(commitCallback)),
       hooks(hooks),
       kv(*db),
+      metadata(*db),
       commitTasks(*this) {
   db->onWrite(KJ_BIND_METHOD(*this, onWrite));
+  // TODO(now): populate lastConfirmedScheduledAlarm asynchronously from alarm manager.
```

Comment:
TODO(now)?

But do we actually need this? We can assume the time in the alarm manager is _no later than_ the time in sqlite. I think if we set `lastConfirmedScheduledAlarm` to the value sqlite, the worst that will happen is we might update the alarm when we didn't need to.

Reply:
Ah, yeah, that makes sense; I can take it out.

I figured there were cases where we might do extra alarm updates when recovering from failures, but I wasn't sure if we maybe also wanted to try to detect "shouldn't happen" cases like where the scheduled alarm time is later than the sqlite state (or unset?).  But maybe those can be detected other ways; we certainly could log the appearances we notice at handler run time.

Reply:
Removed TODO and populated value in 784a55365 .